libs/wingsdk/src/target-tf-aws/topic.ts

```diff
@@ -168,11 +169,9 @@ export class Topic extends cloud.Topic implements IAwsTopic {
   }
 
   public onLift(host: IInflightHost, ops: string[]): void {
-    if (!(host instanceof Function)) {
-      throw new Error("topics can only be bound by tfaws.Function for now");
-    }
-
-    host.addPolicyStatements(...calculateTopicPermissions(this.topic.arn, ops));
+    AwsInflightHost.from(host)?.addPolicyStatements(
+      ...calculateTopicPermissions(this.topic.arn, ops)
+    );
```

Comment:
I wonder if it will be an issue if you try using e.g. an AWS bucket from a GCP function and the user only gets permission errors at deploy time. Something to keep in the back of our minds for later perhaps

Reply:
Mmmm.. Good point. Perhaps `AwsInflightHost.from()` should not throw if the host is not compatible.

Reply:
More defensive.

Reply:
Fixed.